### ISSUE TITLE
DebuggedApplication middleware could display local variables 

### ISSUE THREAD

**Issue by sblondon** [pallets/werkzeug#1309]
I think it would be nice to show local variables for each frame. It would be useful, especially if the debugger is disabled.

For the interface, the local variables could be in a block which would be folded and unfolded with javascript. (This behaviour is equivalent to the django debugger middlware.)

I plan to do a PR if you think it's a good idea.

**Comment by davidism** [comment]
You can already do that by typing `dump()` in a frame's console.

**Comment by RonnyPfannschmidt** [comment]
@davidism i @sblondon meant for when the console is disabled for a debugger (so dump wont work)

**Comment by davidism** [comment]
That seems like a security issue. Tying it to the console means you have to enter a pin before seeing that information.

**Comment by RonnyPfannschmidt** [comment]
@davidism would it be reasonable to have it require the pin as well (i believe not having a python shell would limit the damage somebody that took the credentials could do while still enabling basic debugging)

**Comment by sblondon** [comment]
Yes, It would be a solution: when the pin is validated, each local variable would be inserted in the html.

**Comment by valentijnscholten** [comment]
Are there any plans to add this? I am new to the Werkzeug debugger and I like the console. I know I can get the local variables by using the console, but it would be a lot more convenient if they were there by default.

I noticed that by default the Django error page also shows the local variables when DEBUG is set to True.

**Comment by davidism** [comment]
Given the response to #3244 has been favorable, regardless of if I go ahead with it or not I don't think I want to _extend_ the debugger further. Even back in 2018 when this was opened, I already was not using the Werkzeug debugger UI and instead using PyCharm. VS Code, pudb, and other debuggers are also all capable of quickly interacting with and inspecting the stack.
